publisher-sdk/src/main/java/com/criteo/publisher/CriteoBannerView.java

```diff
@@ -49,6 +59,51 @@
   @Nullable
   private CriteoBannerEventController criteoBannerEventController;
 
+  /**
+   * Used when setting {@link CriteoBannerView} in XML
+   */
+  public CriteoBannerView(@NonNull Context context, AttributeSet attrs) {
+    super(context, attrs);
+    criteo = null;
+
+    TypedArray a = context.getTheme().obtainStyledAttributes(
+        attrs,
+        R.styleable.CriteoBannerView,
+        0,
+        0
+    );
+
+    try {
+      int width = a.getInteger(
+          R.styleable.CriteoBannerView_criteoAdUnitWidth,
+          UNSET_DIMENSION_VALUE
+      );
+      int height = a.getInteger(
+          R.styleable.CriteoBannerView_criteoAdUnitHeight,
+          UNSET_DIMENSION_VALUE
+      );
+      String adUnitId = a.getString(R.styleable.CriteoBannerView_criteoAdUnitId);
+
+      if (adUnitId != null && width != UNSET_DIMENSION_VALUE && height != UNSET_DIMENSION_VALUE) {
+        logger.info("CriteoBannerView inflated for Standalone integration.");
+        bannerAdUnit = new BannerAdUnit(adUnitId, new AdSize(width, height));
+      } else if (adUnitId == null && width == UNSET_DIMENSION_VALUE
+          && height == UNSET_DIMENSION_VALUE) {
+        logger.info("CriteoBannerView inflated for InHouse integration.");
+        bannerAdUnit = null;
+      } else {
+        bannerAdUnit = null;
+        PreconditionsUtil.throwOrLog(new IllegalStateException(
```

Comment:
throwOrLog doesn't throw in production. Is this expected ? Why not throwing directly ?

Reply:
We should never throw in production. Throwing unchecked exceptions is only for debugging purposes, apps should never crash in production due to Criteo.

Reply:
I think we should never throw in production due to variable data coming from publishers or TLA. In this case, the data is constant and the situation is clearly an error. Like when the publisher gives a null cpId, the SDK is crashing during the init to indicate clearly that the integration is wrong.

Reply:
> Like when the publisher gives a null cpId, the SDK is crashing during the init to indicate clearly that the integration is wrong.

It is no longer the case: https://github.com/criteo/android-publisher-sdk/commit/d3f90a182ca314e6f8e30a8fffef850af1156841#diff-ed8fa8b89a351ac154181bc558d65aa2a98aa2b3e4f861c13ffd332f5277025dR119

> In this case, the data is constant and the situation is clearly an error.
Yes, but Criteo isn't essential for the app to function properly. While publishers will miss some revenue due to this, it is not a reason for crashing the app in production. Instead these cases should be caught with logging.

Making an app crash is really extreme. Imagine the impact from an end user perspective: you are on a shopping application and you add an item to the basket.  Then, when going on the basket page, the app crashes because the banner isn't setup properly. How would you react as a user? (I would rate the app 1 star). How would the publisher reacts if it sees that Criteo is one of their top crashes? (as a publisher I would convince the sales team to drop the SDK showing the bad impact it had on the app's rating and buying rates). So crashing in debug mode -> Yes that's good, it enables the publishers to fix an issue before going to prod. Crashing in production? Must not happen under any circumstances in our case. I would of course throw IllegalStateException if the app wasn't in a proper state to continue running properly, but showing an Ad doesn't count as `important to function properly`.

Reply:
I expected publishers to start their app and see if they crash.
But ok to not throw then.

Reply:
Publishers don't have access to the debug mode.
Well, they have if they use `com.criteo.publisher:criteo-publisher-sdk-debug`. But I doubt that anyone is using it.